Fix short forms of weekday and month names

hasso opened the pull request:
This applies the fix already in the core (acc5a7d) to the local copy.
Should fix a weekdays problem mentioned in #476, #891 and possible more.

gvde commented:
I think the short names for days and months need to be translated. Just cutting the full names short is no viable solution for ALL languages.

hasso commented:
These are translated in the core - that's what I fixed in there with commit acc5a7d. In fact I thought that I fixed calendar, but randomly old sliced versions reappeared. I tracked it down to the code this code in calendar.

gvde commented:
Oops. Sorry. My bad. For whatever reason I have just confused the left and right side of that diff...

It's exactly how it should be...
